### ISSUE TITLE
[HDRP] Underwater effect throws ArgumentNullException when switching camera

### ISSUE THREAD

**Issue by daleeidd** [crest-ocean/crest#513]
**Describe the bug**
Underwater effect throws ArgumentNullException when switching back to camera.

**Log**

Only pertinent lines included for brevity:
```
ArgumentNullException: Value cannot be null.
Parameter name: material
Crest.UnderwaterPostProcessUtils.PopulateOceanMask (UnityEngine.Rendering.CommandBuffer commandBuffer, UnityEngine.Camera camera, System.Collections.Generic.List`1[T] chunksToRender, UnityEngine.Plane[] frustumPlanes, UnityEngine.RenderTexture colorBuffer, UnityEngine.RenderTexture depthBuffer, UnityEngine.Material oceanMaskMaterial, System.Boolean debugDisableOceanMask) (at Assets/Crest/Crest/Scripts/Underwater/UnderwaterPostProcessUtils.cs:98)
Crest.UnderwaterPostProcessMaskRenderer.BeginCameraRendering (UnityEngine.Rendering.ScriptableRenderContext scriptableRenderContext, UnityEngine.Camera camera) (at Assets/Crest/Crest/Scripts/Underwater/UnderwaterPostProcessMaskRenderer.cs:48)
```

**Versions**
Crest HDRP 4.2

**To Reproduce**
Switch cameras during play mode. Requires switching back to a previously used camera.

**Additional context**
~~This is a shared state issue where previous camera cleanup is getting called after new camera setup which destroys the material.~~ Nope. The issue is that the material is destroyed and the initialisation step is skipped for the mask renderer so it has no material.

**Comment by banan5555** [comment]
As a workaround I just comment destroy in Cleanup (material leak). Why you need to create and destroy this material every time on switch camera? It will be cheaper to create it once per camera and never destroy - just reuse same material.

**Comment by daleeidd** [comment]
Thanks. We will do something like that. Currently, the material is a shared resource held by the post process effect. Guarding against destroying the material if a camera exists (with a static camera count or something) is probably the solution we will go with.

**Comment by daleeidd** [comment]
@moosichu I am looking at resolving this. Every time a camera switches, the ocean mask material is released by UnderwaterPostProcessHDRP (clean and setup is called). And then when we switch back to that camera, the UnderwaterPostProcessMaskRenderer is already created so we don't initialise it and then it has no ocean mask material (causing the exception).

Are we planning on keeping UnderwaterPostProcessMaskRenderer going forward or are we merging it back in to UnderwaterPostProcessHDRP (built-in doesn't have it yet)?

If we do keep it, should we just destroy the mask renderer too on cleanup? I'm thinking it might be the best way since we don't control the setup/cleanup events.

**Comment by daleeidd** [comment]
Went with destroying the mesh renderer component for now. Fix will be in 4.3

**Comment by daleeidd** [comment]
4.3 is out. Please reopen if you still experience this issue
